### PR TITLE
Utopia 376 & 403 landing page update

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- update modal 

- hide nav bar if user not login
Screen Shot 2022-11-06 at 9 36 17 PM
Screen Shot 2022-11-06 at 9 38 36 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

### CODE REVIEW

**Comment by akroon3r** [comment]
Frontend lint did not pass @binggaofw

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]
Few observations while testing
- Sign out button - design does not match
- Sign out dialog popup does not match designs - buttons and spacing
- After successfully signing out "from Landing Page", the landing page still shows "Access App"

File: src/frontend/src/components/common/Header/Header.tsx [hunk 1]
```diff
@@ -104,13 +104,17 @@ function Header({ user }: Props) {
...
+          confirmLabel="Yes,sign out"
           cancelLabel="Cancel"
+          titleText="Sign out confirmation"
           show={showModal}
           handleClose={hideModalDialog}
           handleCancel={cancelModalDialog}
         >
-          <p>Are you sure you want to logout curren session?</p>
+          <p className="modal-text">
+            Are you sure you would like to sign out of the platform?You will be
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/Header/Header.tsx:115]: Missing space between words `Platform?You`

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 2]
```diff
@@ -1,21 +1,46 @@
 .modal {
-  position: fixed;
+  position: flex;
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/Modal/_modal.scss:2]: `position: flex`? 🤔  Guess that's not even a property, or I am not aware of at least

Comment (binggaofw) [reply src/frontend/src/components/common/Modal/_modal.scss:2]: yes, you are right

File: src/frontend/src/pages/LandingPage/LandingPage.tsx [hunk 3]
```diff
@@ -28,10 +28,19 @@ function LandingPage() {
...
+                Log in with IDIR
+                <FontAwesomeIcon className="icon" icon={faUser} />
+              </button>
+            )}
+            {win.localStorage.getItem('userName') !== null &&
+              win.localStorage.getItem('userName') !== 'undefined' && (
+                <Link className="btn-primary" to="/ppq">
+                  Access App
+                </Link>
+              )}
```
[the remark this answers is not in the document]
Comment (binggaofw) [reply src/frontend/src/pages/LandingPage/LandingPage.tsx:43]: i will create a helper function or custom hook to expose isLoggedIn property, that will make code more clear

**Review by kushal-arora-fw — APPROVED** [review]
3 followup bugs
1. win.localstorage refactor to helper function
2. Sign out button - design does not match
3. After successfully signing out "from Landing Page", the landing page still shows "Access App"

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]
Since it's passed codefreeze; Moving it back to Request changes to get this addressed

1. win.localstorage refactor to helper function
2. Sign out button - design does not match
3. After successfully signing out "from Landing Page", the landing page still shows "Access App"

**Comment by akroon3r** [comment]
@binggaofw @kushal-arora-fw continue to work on these issues but lets hold the merge of this until we officially roll over to sprint 4 .

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 4]
```diff
@@ -5,17 +5,47 @@
...
+    width: 100%;
+    height: 0;
+    border: 2px solid #E2A014;
+    margin-block: 18px 16px;
+  }
+
+   .btn-secondary {
+    margin-top: 0.2em;
+    border: 0;
+   }
```
Comment (Bottle7) [inline src/frontend/src/components/common/Modal/_modal.scss:29]: This affects every button with this class so I suggest nesting a class in the btn-secondary class inside of  _buttons.scss. 

Implementation would look something like 
```
.btn-secondary {
  background-color: $bg-white;
  color: $primary-blue;
  border: 2px solid $primary-blue;
  border-radius: 4px;
  text-align: center;
  padding-block: 12px;
  padding-inline: 25px;
  text-decoration: none;
  display: flex;
  max-width: fit-content;
  white-space: nowrap;
  gap: 0.7em;
  align-items: center;
  justify-content: space-evenly;
  font-weight: 700;
  letter-spacing: 1px;
  cursor: pointer;

  & > .icon {
    font-size: x-large;
  }

  &:hover {
    text-decoration: underline;
    opacity: 0.8;
  }

  &:focus {
    outline: 4px solid $accent-lightblue;
    outline-offset: 1px;
  }

  &:active {
    opacity: 1;
  }

  &.btn-secondary__modal {
     margin-top: 0.2em;
     border: 0;
   }
}
```

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 5]
```diff
@@ -5,17 +5,47 @@
...
+  .modal-title {
+    display: block;
+    font-weight: bold;
+    position: relative;
+    text-align: center;
+    padding-top: 1em;
+    color: black;
+  }
+
+  .horizontal-divider {
```
Comment (Bottle7) [inline src/frontend/src/components/common/Modal/_modal.scss:19]: There is a class already using this name so I would suggest a different name if the current implemented appearance doesn't meet your needs.

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 6]
```diff
@@ -5,17 +5,47 @@
   width: 100%;
   height: 100%;
   background: rgba(0, 0, 0, 0.6);
+  border-radius: 5px;
 
+  .modal-title {
+    display: block;
+    font-weight: bold;
+    position: relative;
```
Comment (Bottle7) [inline src/frontend/src/components/common/Modal/_modal.scss:13]: `position: relative` is used when you want to use `position: absolute` on a child element. Unless I am missing it, there isn't anything being absolutely positioned within this element.

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 7]
```diff
@@ -5,17 +5,47 @@
...
+  }
+
+   .btn-secondary {
+    margin-top: 0.2em;
+    border: 0;
+   }
+
+  .modal-text {
+    display: block;
+    position: relative;
```
Comment (Bottle7) [inline src/frontend/src/components/common/Modal/_modal.scss:33]: Same as above. No `position: absolute` child elements.

File: src/frontend/src/components/common/Modal/_modal.scss (hunk 4 above)
Comment (binggaofw) [reply src/frontend/src/components/common/Modal/_modal.scss:29]: done

File: src/frontend/src/components/common/Modal/_modal.scss (hunk 5 above)
Comment (binggaofw) [reply src/frontend/src/components/common/Modal/_modal.scss]: create a new selectot for modal

File: src/frontend/src/components/common/Modal/_modal.scss (hunk 6 above)
Comment (binggaofw) [reply src/frontend/src/components/common/Modal/_modal.scss]: remove

File: src/frontend/src/components/common/Modal/_modal.scss (hunk 7 above)
Comment (binggaofw) [reply src/frontend/src/components/common/Modal/_modal.scss]: removed

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 8]
```diff
@@ -5,17 +5,40 @@
   width: 100%;
   height: 100%;
   background: rgba(0, 0, 0, 0.6);
+  border-radius: 5px;
+
+  .modal-title {
+    display: block;
+    font-weight: bold;
+    text-align: center;
+    padding-top: 1em;
```
Comment (Bottle7) [inline src/frontend/src/components/common/Modal/_modal.scss:14]:
```suggestion

```

File: src/frontend/src/components/common/Modal/_modal.scss [hunk 9]
```diff
@@ -5,17 +5,40 @@
...
+  }
 
   .modal-main {
-    position: fixed;
-    background-color: $link-blue;
-    width: 60%;
+    position: relative;
+    background-color: #fff;
+    width: 35%;
     height: auto;
```
Comment (Bottle7) [inline src/frontend/src/components/common/Modal/_modal.scss:36]:
```suggestion
    height: auto;
    padding-block: 1.5em;
    padding-inline: 1em;
```

**Review by Bottle7 — APPROVED** [review]

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**